gromacs/fileformats/top.py

```diff
@@ -0,0 +1,1243 @@
+        top = top.replace('*ATOMTYPES*',      ''.join( self._make_atomtypes(self.system)) )
+        top = top.replace('*NONBOND_PARAM*',  ''.join( self._make_nonbond_param(self.system)) )
+        top = top.replace('*PAIRTYPES*',      ''.join( self._make_pairtypes(self.system)) )
+        top = top.replace('*BONDTYPES*',      ''.join( self._make_bondtypes(self.system)) )
+        top = top.replace('*CONSTRAINTTYPES*',''.join( self._make_constrainttypes(self.system)))
+        top = top.replace('*ANGLETYPES*',     ''.join( self._make_angletypes(self.system)))
+        top = top.replace('*DIHEDRALTYPES*',  ''.join( self._make_dihedraltypes(self.system)) )
+        top = top.replace('*IMPROPERTYPES*',  ''.join( self._make_impropertypes(self.system)) )
+        top = top.replace('*CMAPTYPES*',      ''.join( self._make_cmaptypes(self.system)) )
+
+
+        for i,(molname,m) in enumerate(self.system.dict_molname_mol.items()):
+
+            itp = self.itptemplate
+            itp = itp.replace('*MOLECULETYPE*',  ''.join( self._make_moleculetype(m, molname))  )
+            itp = itp.replace('*ATOMS*',         ''.join( self._make_atoms(m))  )
+            itp = itp.replace('*BONDS*',         ''.join( self._make_bonds(m))  )
+            itp = itp.replace('*PAIRS*',         ''.join( self._make_pairs(m))  )
+            itp = itp.replace('*SETTLES*',       ''.join( self._make_settles(m))  )
+            itp = itp.replace('*VIRTUAL_SITES3*',''.join( self._make_virtual_sites3(m))  )
+            itp = itp.replace('*EXCLUSIONS*',    ''.join( self._make_exclusions(m))  )
+            itp = itp.replace('*ANGLES*',        ''.join( self._make_angles(m)) )
+            itp = itp.replace('*DIHEDRALS*',     ''.join( self._make_dihedrals(m)) )
+            itp = itp.replace('*IMPROPERS*',     ''.join( self._make_impropers(m)) )
+            itp = itp.replace('*CMAPS*',         ''.join( self._make_cmaps(m)) )
+            if not self.multiple_output:
+                top += itp
+            else:
+                outfile = "mol_{}.itp".format(molname)
+                top += '#include "mol_%s.itp" \n' % molname                
+                with open(outfile, "w") as f:
+                    f.writelines([itp])
+
+        top += '\n[system]  \nConvertedSystem\n\n'
+        top += '[molecules] \n'
+        molecules = [("", 0)]
+
+        for m in self.system.molecules:
+            if (molecules[-1][0] != m.name): 
+                molecules.append([m.name, 0])
+            if molecules[-1][0] == m.name:
+                molecules[-1][1] += 1
+
+        for molname, n in molecules[1:]:
+            top += '%s     %s\n' % (molname, n)
+        top += '\n'
+
+        with open(self.outfile, 'w') as f:
+            f.writelines([top])
+
+        return
+        for i, m in enumerate(self.system.molecules):
+            molname = 'mol_%02d' % (i+1)
+            top += '#include "itp_%s.itp" \n' % molname
+
```

Comment:
why all the white space?? (Generally, I'd cut down on empty lines and only do 1 line between functions/methods, 2 lines between classes)